crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -24,6 +24,18 @@ impl Direction2d {
         Self::new_and_length(value).map(|(dir, _)| dir)
     }
 
+    /// Create a [`Direction2d`] from a [`Vec2`] that is already normalized.
+    ///
+    /// ## Safety
+    ///
+    /// This function only checks if `value` is normalized on debug builds,
+    /// release builds can allow the creation of a [`Direction2d`] that is not normalized.
```

Comment:
```suggestion
    /// `value` must be normalized, i.e it's length must be `1.0`.
```
I think it's better to have something more straightforward for the safety section.